app/components/resource_edit_button.rb

```diff
@@ -1,13 +1,6 @@
 # frozen_string_literal: true
 
-class EditResourceButton < ApplicationComponent
-  attr_reader :resource, :policy
-
-  def initialize(resource:, policy:)
-    @resource = resource
-    @policy = policy
-  end
-
+class ResourceEditButton < ResourceButton
   def path
```

Comment:
@awead - we are using `policy` in the `path` method of `ResourceEditButton`:
```
elsif policy.new?
  # create a new draft work version
  dashboard_work_work_versions_path(resource.work)
end
```
Though I wonder if we can either derive it from within the component instead of passing it in, or if we even need it at all, since the app would probably 403 or something if the user tried to create a new draft version and they somehow did not have the permissions to do so. As it is now, if they somehow didn't have permissions to create a new draft, the link would be incorrect, so would kind of be a different bug.

Maybe we can just change the code to an `else` only, and stop passing in `policy`?

```
else
  # create a new draft work version
  dashboard_work_work_versions_path(resource.work)
end
```

Let me know what you think.

Reply:
@nicktiberi No, I like being explicit with `.new?`. I was just missing it.